File: SecondYearCS/OOP/CivitasRB/lib/CasillaCalle.rb
require_relative 'sorpresa.rb'
require_relative 'casilla.rb'
require_relative 'tipo_sorpresa.rb'
require_relative 'titulo_propiedad.rb'
require_relative 'mazo_sorpresa.rb'
require_relative 'diario.rb'
require_relative 'dado.rb'
module CivitasRB
  class CasillaCalle < Casilla
    def initialize(tituloPropiedad,nombre)
      super(nombre)
      @tituloPropiedad = tituloPropiedad
    end

    def recibeJugador(act, todos)
      if self.jugadorCorrecto(act, todos)
        self.informe(act, todos)
        jugadorActual = todos.at(act)
        if !@tituloPropiedad.tienePropietario
          if jugadorActual.puedeComprarCasilla
            @tituloPropiedad.tramitarAlquiler(jugadorActual)
          end
        end
      end
    end

    def getTituloPropiedad
      return @tituloPropiedad
    end


    def to_s
      res = super.to_s
      if (@tituloPropiedad != nil)
        res += " con: " + "\n 1- precioCompra: " + @tituloPropiedad.precioCompra.to_s +
            "\n 2- precioEdificación: " + @tituloPropiedad.precioEdificar.to_s +
            "\n 3- AlquilerBase: " + @tituloPropiedad.alquilerBase.to_s
      end
      return res
    end

  end
end

File: SecondYearCS/OOP/Civitas/src/civitas/CasillaCalle.java
package civitas;
import java.util.ArrayList;

public class CasillaCalle extends Casilla {

   
    private TituloPropiedad tituloPropiedad; // tipo:: CALLE

    //Tipo Calle
    CasillaCalle(TituloPropiedad titulo, String nombre){
        super(nombre);
        this.tituloPropiedad = titulo;    
    }
   
    @Override
    void recibeJugador(int actual, ArrayList<Jugador> todos){
        if(this.jugadorCorrecto(actual,todos)){
            this.informe(actual,todos);
            Jugador jugadorActual = todos.get(actual);
            if(!this.tituloPropiedad.tienePropietario()){
                if(jugadorActual.puedeComprarCasilla())
                    this.tituloPropiedad.tramitarAlquiler(jugadorActual);
            }
        }
    }
    
    @Override
    public String toString(){
       String result = super.toString();
       result += "\n 1- precio de compra: " + this.tituloPropiedad.getPrecioCompra()
               + "\n 2- precio de edificación: " + this.tituloPropiedad.getPrecioEdificar()
               + "\n 3- alquiler base: " + this.tituloPropiedad.getAlquilerBase();     
       return result;
    }

    public TituloPropiedad getTituloPropiedad(){
        return this.tituloPropiedad;
    }  
}
